### ISSUE TITLE
Hard to use API in non-panicing programs

### ISSUE THREAD

**Issue by rbtcollins** [rory/rust-dbf#3]
It would be easier to use if you surfaced Result's rather than unwrapping errors - then folk using the library wouldn't need to use catch_unwind...

**Comment by amandasaurus** [comment]
Yeah good idea.

**Comment by rbtcollins** [comment]
I can create a PR if you like.

**Comment by amandasaurus** [comment]
Yes please, I'm a little busy and won't have much time to look at this myself in the immediate future

On 5 June 2018 23:15:10 CEST, Robert Collins <notifications@github.com> wrote:
>I can create a PR if you like.

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
